com.unity.render-pipelines.universal/ShaderLibrary/ShaderDebugPrint.hlsl

```diff
@@ -0,0 +1,279 @@
+#ifndef SHADER_DEBUG_PRINT_INCLUDED
+#define SHADER_DEBUG_PRINT_INCLUDED
+
+// Include this header to any shader to enable debug printing values from shader code to console
+//
+// Example:
+// float4 colorRGBA = float4(0.1, 0.2, 0.3, 0.4);
+// if(all(int2(pixel.xy) == int2(100, 100))
+//     ShaderDebugPrint(ShaderDebugTag('C','o','l'), colorRGBA);
+// ----
+// Output:
+// Frame #270497: Col  float4(0.1f, 0.2f, 0.3f, 0.4f)
+
+// Output buffer bound into "last" slot by convention
+RWStructuredBuffer<uint> shaderDebugOutputData : register(u7);
```

Comment:
Do we need to explicitly specify the slot here?
What happens on a VK device that has 4 UAV max (which, if I remember correctly, is the guaranteed min spec)?

Reply:
Ideally no (that's the goal).
But we didn't find a way to affect that programmatically and it would be something that doesn't conflict with any existing shader usage (so that it's automatically useable from any shader). By convention "last" available would be a good guess.
7 here is just a hand-wavy hack as we didn't find a way to control this from code automatically.